src/gradient/sensitivity_calculator.py

```diff
@@ -60,7 +70,12 @@ def run(self, scale: str, params: dict):
         self._calculate_ngm_gradients()
 
         # 7. Calculate eigenvalue and eigenvalue gradients for R0
-        self._calculate_eigenvectors()
+        if self.method == "eig":
```

Comment:
Remove this `if` and `self._calculate_eigenvectors`, since it is mathematically incorrect to calculate the eigenvector as we have done previously.